### PR TITLE
Fix sanitizer findings

### PR DESCRIPTION
**Pull request by ligurio**
Changes makes OSS-Fuzz CI green - https://github.com/google/oss-fuzz/pull/10448

### CODE REVIEW

File: libfreerdp/common/assistance.c [hunk 1]
```diff
@@ -704,7 +704,7 @@ static BOOL freerdp_assistance_decrypt2(rdpAssistanceFile* file, const char* pas
 BYTE* freerdp_assistance_hex_string_to_bin(const void* raw, size_t* size)
 {
 	BYTE* buffer = NULL;
-	size_t length, rc;
+	size_t length = 0, rc = 0;
```
Comment (akallabeth) [inline libfreerdp/common/assistance.c:707]: false positive, better move down below as `const size_t length = strlen...`, same for rc

File: libfreerdp/common/test/TestFuzzCommonAssistanceParseFileBuffer.c [hunk 2]
```diff
@@ -4,12 +4,21 @@ static int parse_file_buffer(const uint8_t* Data, size_t Size)
 {
 	static const char TEST_MSRC_INCIDENT_PASSWORD_TYPE2[] = "48BJQ853X3B4";
 	int status = -1;
-	rdpAssistanceFile* file = freerdp_assistance_file_new();
+	rdpAssistanceFile* file = NULL;
```
Comment (akallabeth) [inline libfreerdp/common/test/TestFuzzCommonAssistanceParseFileBuffer.c:7]: what is this change for?

File: libfreerdp/common/test/TestFuzzCommonAssistanceParseFileBuffer.c [hunk 3]
```diff
@@ -4,12 +4,21 @@ static int parse_file_buffer(const uint8_t* Data, size_t Size)
...
+	file = freerdp_assistance_file_new();
 	if (!file)
 		return -1;
-	status = freerdp_assistance_parse_file_buffer(file, (char*)Data, Size,
+
+	char *buf = calloc(Size + 1, sizeof(char));
+	if (buf == NULL)
+		return 0;
```
Comment (akallabeth) [inline libfreerdp/common/test/TestFuzzCommonAssistanceParseFileBuffer.c:14]: leaking `file`

**Review by akallabeth — CHANGES_REQUESTED** [review]

File: libfreerdp/common/assistance.c (hunk 1 above)
Comment (ligurio) [reply libfreerdp/common/assistance.c]: Fixed.

File: libfreerdp/common/test/TestFuzzCommonAssistanceParseFileBuffer.c (hunk 2 above)
Comment (ligurio) [reply libfreerdp/common/test/TestFuzzCommonAssistanceParseFileBuffer.c]: Reverted.

**Review by akallabeth — APPROVED** [review]
LGTM

**Comment by akallabeth** [comment]
@freerdp-bot test please

**Comment by ligurio** [comment]
Fixed code style.

```diff
--- a/libfreerdp/common/test/TestFuzzCommonAssistanceHexStringToBin.c
+++ b/libfreerdp/common/test/TestFuzzCommonAssistanceHexStringToBin.c
@@ -2,7 +2,7 @@
 
 int LLVMFuzzerTestOneInput(const uint8_t* Data, size_t Size)
 {
-	char *buf = calloc(Size + 1, sizeof(char));
+	char* buf = calloc(Size + 1, sizeof(char));
 	if (buf == NULL)
 		return 0;
 	memcpy(buf, Data, Size);
diff --git a/libfreerdp/common/test/TestFuzzCommonAssistanceParseFileBuffer.c b/libfreerdp/common/test/TestFuzzCommonAssistanceParseFileBuffer.c
index 063a943eb..190097771 100644
--- a/libfreerdp/common/test/TestFuzzCommonAssistanceParseFileBuffer.c
+++ b/libfreerdp/common/test/TestFuzzCommonAssistanceParseFileBuffer.c
@@ -8,7 +8,7 @@ static int parse_file_buffer(const uint8_t* Data, size_t Size)
 	if (!file)
 		return -1;
 
-	char *buf = calloc(Size + 1, sizeof(char));
+	char* buf = calloc(Size + 1, sizeof(char));
 	if (buf == NULL)
 		goto err;
 	memcpy(buf, Data, Size);
```

**Pull request merged**